lib/active_shipping/carriers/canada_post_pws.rb

```diff
@@ -564,6 +578,36 @@ def log(msg)
 
     private
 
+    def contract?
+      @options[:contract_id].present?
+    end
+
+    SHIPMENT_NON_CONTRACT_MIMETYPE = "application/vnd.cpc.ncshipment+xml"
```

Comment:
The latest version for non-contract shipping is version 4, see https://www.canadapost.ca/cpo/mc/business/productsservices/developers/services/onestepshipping/createshipment.jsf

I can provide you with login credentials.

Otherwise :+1:  on the helper method.